### PR TITLE
fix: sd-navigation-item a11y

### LINKED ISSUE #1481: feat[dev]: ✨ implement A11y improvements to sd-navigation-item
## Description
This task describes the Storybook implementation for the sd-navigation-item component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-navigation-item can be found here: [sd-navigation-item Notes](https://www.notion.so/matuzo/sd-navigation-item-dc889e6da8ed49d3932deba051ffe20e).

-------------------------

## Comment
**Problem:** Accordion items don't provide enough semantic information.
**Solution:** Add and toggle aria-expanded 

```
<button aria-expanded="false" aria-controls="navigation-item-details">
    Navigation
</button>
<div id="navigation-item-details" hidden>
    …
</div>
```

**Problem:** aria-controls on accordion items points to a non-existent id ("navigation-item-details").
**Solution:** Fix the relation or remove the attribute.

**Problem:** The accessible name of the button in the “separated” example is "chevron down button".
**Solution:** Provide a label that describes the button's purpose.

## Notes
- Disabled navigation items: See [@sd-button](https://www.notion.so/sd-button-104f3f6e1bc0800b9f72d2a046316ec3?pvs=24). Everything I wrote there about disabling form elements applies here as well.
- Labels of buttons and links should be descriptive but concise. Items with descriptions create long labels.

## Best Practices
- Use links for navigation and buttons for performing other actions 

## Related WCAG criteria
- https://www.w3.org/WAI/WCAG22/quickref/#name-role-value

### Open Questions
- [ ] Question1
- [ ] Question2

## DoR
- [ ] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### PR DESCRIPTION
**Pull request by Vahid1919**
## Description:

Closes #1481. 

## Definition of Reviewable:

- [x] Documentation is created/updated
- [x] E2E tests (features, a11y, bug fixes) are created/updated

- [x] relevant tickets are linked

### CODE REVIEW

File: packages/components/src/components/navigation-item/navigation-item.ts [hunk 1]
```diff
@@ -222,7 +222,7 @@ export default class SdNavigationItem extends SolidElement {
               ? this.separated
                 ? html`<button
                     type="button"
-                    title="chevron-down-button"
+                    title="toggle-accordion"
```
Comment (karlbaumhauer) [inline packages/components/src/components/navigation-item/navigation-item.ts:225]:
```suggestion
                    title="toggle-details"
```

Comment (karlbaumhauer) [reply packages/components/src/components/navigation-item/navigation-item.ts:225]: I am not 100% aware of the usecase for this title, but as we used summary and details down in the code, it kind of makes it strange to talk about accordions here?!?

Or am I missing something?

Comment (karlbaumhauer) [reply packages/components/src/components/navigation-item/navigation-item.ts:225]: we could also rename it with the actual purpose. As this is nothing generic but the navigation-item, we could also call it "toggle-navigation-details"

**Review by karlbaumhauer — CHANGES_REQUESTED** [review]

File: packages/components/src/components/navigation-item/navigation-item.ts (hunk 1 above)
Comment (smfonseca) [reply packages/components/src/components/navigation-item/navigation-item.ts:225]: Second that

**Review by karlbaumhauer — APPROVED** [review]

**Pull request merged**